src/pages/profile/index.tsx

```diff
@@ -39,14 +39,35 @@ profile.get<"/:handle">(async (c) => {
   const contStr = c.req.query("cont");
   const cont = contStr == null || contStr.trim() === "" ? undefined : contStr;
   if (cont != null && !isUuid(cont)) return c.notFound();
+  const pageStr = c.req.query("page");
+  if (
+    pageStr !== undefined &&
+    (Number.isNaN(Number.parseInt(pageStr)) || Number.parseInt(pageStr) < 1)
+  ) {
+    return c.notFound();
+  }
+  const page =
+    pageStr !== undefined && !Number.isNaN(Number.parseInt(pageStr))
+      ? Number.parseInt(pageStr)
+      : 1;
+  const totalPosts = await db.query.posts.findMany({
+    where: and(
+      eq(posts.accountId, owner.id),
+      or(eq(posts.visibility, "public"), eq(posts.visibility, "unlisted")),
+    ),
+  });
+  const maxPage = Math.ceil(totalPosts.length / PAGE_SIZE);
+  if (page > maxPage) {
+    return c.notFound();
+  }
```

Comment:
Error that occurs when `?page` exceeds the total number of pages.

![404 not founds screenshots](https://github.com/user-attachments/assets/15348c9a-f17a-4e06-a94e-036114cef694)

Reply:
I think it's fair enough.